modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/MuleSdkThisExtensionModelLoader.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.mule.internal.loader;
+
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_THIS_LOADER_ID;
+
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+import org.mule.runtime.module.extension.internal.loader.AbstractExtensionModelLoader;
+import org.mule.runtime.module.extension.internal.loader.parser.ExtensionModelParserFactory;
+import org.mule.runtime.module.extension.mule.internal.loader.parser.MuleSdkThisExtensionModelParserFactory;
+
+/**
+ * {@link ExtensionModelLoader} implementation for Mule SDK Extensions defined as part of the same artifact.
+ *
+ * @since 4.5.0
+ */
+public class MuleSdkThisExtensionModelLoader extends AbstractExtensionModelLoader {
```

Comment:
This name is worst than the original. Why not simply `ApplicationExtensionModelLoader` ?

Reply:
I just changed this as discussed yesterday. Now we have one `MuleSdkApplicationExtensionModelLoader` and one `MuleSdkExtensionExtensionModelLoader`.
The `MuleSdkApplicationExtensionModelLoader` is the easiest one because it keeps doing what it has been doing before.
The `MuleSdkExtensionExtensionModelLoader` has a decorator for providing the temporary local `ExtensionModel` used for AST validations (in the context of the chicken and egg problem). This way we avoid the complexity of having to deal with different parsers depending on the context and it prepares the ground for the next move which would be generating the `ExtensionModel` only once.
The undecorated one is readily available so it would make no sense to generate it again, the problem now is only that we have it inside the parser, which has no way of letting the loader know it doesn't need to call the `LoaderDelegate` and then pass it to the `ExtensionModelFactory`. I'll experiment with a solution for that now.